grommet/grommet
# Add ability to style the Clear option background and color on hover, in the Select component

#### What does this PR do?
* Allows consumers to style the hover state of the Clear option inside the `Select` component via `select.clear.hover.background` and `select.clear.hover.color` in the theme
    ![clear-option](https://user-images.githubusercontent.com/22405692/138547166-60aa4800-26b5-4f57-8743-ac9363d80f59.gif)
* Update Select stories for #5735
* Adds a custom themed story for the `Select`'s Clear option bottom positioning

#### Where should the reviewer start?
* Run storybook in this branch and view these stories:
    - http://localhost:9001/?path=/story/input-select-clear--top
    - http://localhost:9001/?path=/story/input-select-clear--bottom
* Update the values for `select.clear.hover.background` and `select.clear.hover.color` in the theme, and verify that the hover state gets styled accordingly.


#### What testing has been done on this PR?
Manual tests via Storybook

#### How should this be manually tested?
* Run storybook in this branch and view these stories:
    - http://localhost:9001/?path=/story/input-select-clear--top
    - http://localhost:9001/?path=/story/input-select-clear--bottom
* Update the values for `select.clear.hover.background` and `select.clear.hover.color` in the theme, and verify that the hover state gets styled accordingly.

#### Any background context you want to provide?
My team needs to style the Clear option so that it looks consistent with the other `Select` options.

#### What are the relevant issues?
Currently unable to style the `Select` Clear option's hover state.

#### Do the grommet docs need to be updated?
Yes. Theming documentation for `select.clear.hover.background` and `select.clear.hover.color` need to be added to https://v2.grommet.io/select#theme

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
This enhancement is backward-compatible.
